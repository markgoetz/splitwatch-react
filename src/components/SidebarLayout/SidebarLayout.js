import React from 'react';
import styles from './SidebarLayout.module.css';

export default function SidebarLayout(props) {
    return (<div className={styles.layout}>
        <div className={styles.left}>{ props.children[0] }</div>
        <div className={styles.right}>{ props.children[1] }</div>
    </div>);
}