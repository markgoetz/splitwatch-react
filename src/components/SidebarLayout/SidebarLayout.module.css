.layout {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.right {
  margin-top: 20px;
}

@media (min-width: 769px) {
  .layout {
    flex-direction: row;
  }

  .left {
    flex-basis: 75%;
  }

  .right {
    margin-top: 0;
    margin-left: 40px;
    flex-basis: 25%;
  }
}
