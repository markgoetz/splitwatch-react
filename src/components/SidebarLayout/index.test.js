import React from 'react';
import SidebarLayout from './SidebarLayout';
import { shallow } from 'preact-render-spy';

describe('Sidebar Layout Component', () => {
  test('renders correctly with two children', () => {
    const context = shallow(<SidebarLayout>
      <div>Test 1!</div>
      <div>Test 2!</div>
    </SidebarLayout>);
    expect(context).toMatchSnapshot();
  });
});