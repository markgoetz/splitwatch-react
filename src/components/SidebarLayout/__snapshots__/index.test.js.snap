// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Layout Component renders correctly with two children 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <div>
    <div>Test 1!</div>
  </div>
  <div>
    <div>Test 2!</div>
  </div>
</div>

`;
